client/state/sites/domains/test/reducer.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import domainsReducer, {
+	items as itemsReducer,
+	requesting as requestReducer,
+	errors as errorsReducer
+} from '../reducer';
+
+/**
+ * Action types constantes
+ */
+import {
+	SITE_DOMAINS_RECEIVE,
+	SITE_DOMAINS_REQUEST,
+	SITE_DOMAINS_REQUEST_SUCCESS,
+	SITE_DOMAINS_REQUEST_FAILURE,
+} from 'state/action-types';
+
+/**
+ * Actions creators functions
+ */
+import {
+	createDomainsRequestAction,
+	createDomainsRequestSuccessAction,
+	createDomainsRequestFailureAction
+} from '../actions';
+
+/**
+ * Fixture data
+ */
+import {
+	SITE_ID_FIRST as siteId,
+	SITE_ID_FIRST as firstSiteId,
+	SITE_ID_SECOND as secondSiteId,
+	SITE_FIRST_DOMAINS as siteDomains,
+	DOMAIN_PRIMARY as firstDomain,
+	DOMAIN_NOT_PRIMARY as secondDomain,
+	ERROR_MESSAGE_RESPONSE as errorMessageResponse
+} from './fixture';
+
+/**
+ * Make a generic expect test comparing states before and after
+ * to execute the domains reducer function.
+ *
+ * @param {Object} before - state before instance
+ * @param {Object} action - reducer action
+ * @param {Object} after - state after instance
+ * @param {Function} [reducer] - reducer to test. domainsReducer as default
+ */
+const expectBeforeEqualToAfter = ( before, action, after, reducer = domainsReducer ) => {
```

Comment:
Did this help much? I think it makes the tests a bit harder to read. Was the following too many lines?
```javascript
const originalState = deepFreeze( original );
const newState = reducer( originalState, action );
expect( newState ).to.eql( expectedState );
```